src/Features/CustomerEffortScoreTracks.php

```diff
@@ -40,17 +40,44 @@ class CustomerEffortScoreTracks {
 	 */
 	const PRODUCT_UPDATE_ACTION_NAME = 'product_update';
 
+	/**
+	 * Action name for settings change.
+	 */
+	const SETTINGS_CHANGE_ACTION_NAME = 'settings_change';
+
 	/**
 	 * Label for the snackbar that appears when a user submits the survey.
 	 *
 	 * @var string
 	 */
 	private $onsubmit_label;
 
+
 	/**
 	 * Constructor. Sets up filters to hook into WooCommerce.
 	 */
 	public function __construct() {
+		add_action(
+			'admin_init',
+			array(
+				$this,
+				'maybe_clear_ces_tracks_queue',
+			)
+		);
+
+		$this->add_actions();
+	}
+
+	/**
+	 * Add actions that require woocommerce_allow_tracking.
+	 */
+	private function add_actions() {
+		// Only enqueue a survey if tracking is allowed.
+		$allow_tracking = 'yes' === get_option( 'woocommerce_allow_tracking', 'no' );
```

Comment:
I like this optimization! No need to set up the action handlers if we aren't ever going to do anything in them. Nice!